specs/LocaleRegion.md

```diff
@@ -0,0 +1,111 @@
+LocaleRegion
+===
+
+# Background
+Developers have requested access to programmatically choose the specific locale/region to use in WebView2. 
+The locale/region updates the JavaScript region (objects in the ECMA standard) which gets reflected as
+variations in time and date formats. Currently, the locale is by default set to the same value as the 
+display language. 
+
+You can use the `LocaleRegion` API to update the locale value individually from the display
+language.
+
+# Description
+* The intended value for `LocaleRegion` is in the format of `language[-country]` where `language` is the 
+2-letter code from [ISO 639](https://www.iso.org/iso-639-language-codes.html) and `country` is the 
+2-letter code from [ISO 3166](https://www.iso.org/standard/72482.html).
+
+# Examples
+```cpp
+auto webViewEnvironment10 = m_webViewEnvironment.try_query<ICoreWebView2Environment10>();
+wil::com_ptr<ICoreWebView2ControllerOptions> options;
+HRESULT hr = webViewEnvironment10->CreateCoreWebView2ControllerOptions(&options);
+options->put_LocaleRegion(m_webviewOption.localeRegion.c_str());
+```
+
+```c#
+CoreWebView2Environment environment;
+CoreWebView2ControllerOptions CreateCoreWebView2ControllerOptions(CoreWebView2Environment environment)
+{
+    CoreWebView2ControllerOptions ControllerOptions = null;
+    if (LocaleRegion != null)
+    {
+        ControllerOptions = environment.CreateCoreWebView2ControllerOptions();
+        ControllerOptions.LocaleRegion = LocaleRegion;
+    }
+    return ControllerOptions;    
+}
+```
+
+# API Details
+```cpp
+[uuid(0c9a374f-20c3-4e3c-a640-67b78a7e0a48), object, pointer_default(unique)]
+interface ICoreWebView2StagingControllerOptions : IUnknown {
+  /// The default region for the WebView2.  It applies to JavaScript API
+  /// Intl.DateTimeFormat() which affects string formatting like
+  /// in the time/date formats. The intended locale value is in the format of
+  /// `language-country` or `language_country` where `language` is the 2-letter code from [ISO
+  /// 639](https://www.iso.org/iso-639-language-codes.html) and `country` is the
+  /// 2-letter code from [ISO 3166](https://www.iso.org/standard/72482.html).
+  ///
+  /// This property sets the region for a CoreWebView2Environment during its creation. 
+  /// Creating a new CoreWebView2Environment object that connects to an already running 
+  /// browser process cannot change the region previously set by an earlier CoreWebView2Environment.  
+  /// The CoreWebView2Environment and all associated webview2 objects will need to closed.
+  ///
+  /// The default value for LocaleRegion will be depend on the WebView2 language
+  /// and OS region. If the language portions of the WebView2 language and OS Region

```

Comment:
"the language portion... of the OS region" doesn't make sense. Region != language.

Reply:
The locale region format consists of the language code and country code. i.e. for "en-US", the language code is "en".

Reply:
A region is a geographic area. A language is what you speak. I can speak "American English" but live in France, for example. So my language would be "en-us" but my region would be France. Windows separates these two concepts.

Reply:
Note that ECMA is a bit confused on the difference between "language" and "locale", not having a great way of saying "Canadian English speaker in the US".  The beginning of this spec says it's intended for date/time formats.  Well, do they expect en-CA formats or en-US formats?

Reply:
I don't understand why this differs from modern app defaults?  Modern apps select the language from the language profile, and the locale from that.  We already have a ridiculous number of mixed-up language/locale/region paradigms.  Webview shouldn't be reinventing the wheel.  Grab what we're already using.

(It's also horrible that we confuse "UI language" (meaning system apps) with "language profile", but that's out of scope here.)

Reply:
@ShawnSteele  Setting the LocaleRegion always to the OS region in the Control Panel would cause breaking changes. Especially in the cases where the display language is completely different than their OS region.

Reply:
@ptorr-msft @ShawnSteele 
Is there any naming convention you had in mind to describe the value in this control panel setting that I used to describe as "OS Region"? This is what we use to compare with the WebView2 language in order to decide what the default value is.
<img width="694" alt="image" src="https://user-images.githubusercontent.com/24627278/206593605-5e768d4b-693e-43f7-a7b2-1b5438b72cf3.png">